### ISSUE TITLE
./lib/logthrdestdrv.h:33:16: fatal error: iv.h: No such file or directory

### ISSUE THREAD

**Issue by shellscriptcode** [syslog-ng/syslog-ng#1307]
# syslog-ng

## Version of syslog-ng

```
syslog-ng-3.8.1
```

## Platform

Redhat Linux Enterprise Server 6.8

# Issue

## Failure

I am receiving below error message

```
  YACC     modules/add-contextual-data/add-contextual-data-grammar.c
make --no-print-directory all-recursive
  CC       modules/native/modules_native_libsyslog_ng_native_connector_a-native-grammar.o
In file included from lib/cfg-grammar.y:41:0,
                 from modules/native/native-grammar.y:39:
./lib/logthrdestdrv.h:33:16: fatal error: iv.h: No such file or directory
compilation terminated.
make[2]: *** [modules/native/modules_native_libsyslog_ng_native_connector_a-native-grammar.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

```
$ gdb syslog-ng
 run
```

## Steps to reproduce

1. ./configure --prefix=/path && make

2. ...

## Configuration

Compiling syslon-ng-3.8.1 from source

## Input and output logs (if possible)

N/A

**Comment by bazsi** [comment]
again, this indicates a missing libivykis, but that's embedded in the
release tarball. I am not sure how you can start compiling as this should
make the configure script to error out.


-- 
Bazsi

**Comment by lbudai** [comment]
@shellscriptcode : you can download the release tarball for 3.8.1 from [here](https://github.com/balabit/syslog-ng/releases/download/syslog-ng-3.8.1/syslog-ng-3.8.1.tar.gz)

**Comment by Kokan** [comment]
I am closing due to the inactivity.

**Comment by DerDakon** [comment]
I was able to reproduce this with 3.18.1. It's quite simple: keep rebuilding it with "-j 10" on an Quad-Core+HT system. Every once in a while you will see exactly this error. The point is that iv.h is generated in some configure script, but there somewhere is a dependency missing.

**Comment by Kokan** [comment]
@DerDakon thanks for reporting it, I reopened this issue.

**Comment by lbudai** [comment]
hint: maybe there is a race cond. in the dependecy handling in our build files.

**Comment by lbudai** [comment]
( @kokan: we have an internal perf. machine: we could try to build syslog-ng with `-j 10` )

**Comment by Kokan** [comment]
The *native* module clearly does not depend on anything (at least not on ivykis); but I am kinda want to question the existance of that module: https://github.com/Kokan/syslog-ng/commit/1afabfd168b5f7da2a3439171cf21e8ad09ec2b6 (but yesterday got lazy to open the PR)

**Comment by DerDakon** [comment]
I have spent 2 hours on my 8-core machine with -j10 now trying to reproduce this, with no success. So it seems to be fixed in 3.24.1. I see iv.h being created rather early in the build process and installed at the end, so I hope everything is well now?
